node/src/main/scala/coop/rchain/node/configuration/Configuration.scala

```diff
@@ -44,6 +44,8 @@ object Configuration {
   private val DefaultKeyFileName                = "node.key.pem"
   private val DefaultSecureRandomNonBlocking    = false
   private val DefaultMaxNumOfConnections        = 500
+  private val DefaultMaxMessageSize: Int        = 100 * 1024 * 1024
+  private val DefaultThreadPoolSize: Int        = 4000
```

Comment:
That's a lot of threads! :) What's the motivation for a number this high?

Why not use `2 * java.lang.Runtime.getRuntime().availableProcessors()` by default? This way we can load all cores without causing too much context switching.

Reply:
We are actually LIMITING number of threads , because normally scheduler will take all that your system got, killing it at some point (4k threas for mac)

Reply:
I think what @ArturGajowy meant was to have
```scala
private val DefaultThreadPoolSize: Int        = 2 * java.lang.Runtime.getRuntime().availableProcessors()
```
instead of 4000.

Reply:
Ok, fair enough? Why 2 * cores? And not `n`?